extension/src/time_series/pipeline/aggregation.rs

```diff
@@ -410,6 +412,213 @@ extension_sql!(r#"
 ALTER FUNCTION "arrow_pipeline_then_num_vals" SUPPORT toolkit_experimental.pipeline_num_vals_support;
 "#);
 
+pg_type! {
+    #[derive(Debug)]
+    struct PipelineThenCounterAgg<'input> {
+        num_elements: u64,
+        elements: [Element; self.num_elements],
+    }
+}
+
+ron_inout_funcs!(PipelineThenCounterAgg);
+
+#[pg_extern(immutable, parallel_safe, schema="toolkit_experimental")]
+pub fn run_pipeline_then_counter_agg<'s, 'p>(
+    mut timeseries: toolkit_experimental::TimeSeries<'s>,
+    pipeline: toolkit_experimental::PipelineThenCounterAgg<'p>,
+) -> CounterSummary<'static> {
+    timeseries = run_pipeline_elements(timeseries, pipeline.elements.iter());
+    if timeseries.num_points() == 0 {
+        panic!("Unable to create a counter summary from 0 points.");
+    }
```

Comment:
I think this should just return `NULL` instead

```suggestion
    if timeseries.num_points() == 0 {
        return None
    }
```
if there are no relevant elements in a timeseries that shouldn't bring down the entire query.

Reply:
there shouldn't be